tensorflow/core/kernels/resize_bilinear_op_gpu.cu.cc

```diff
@@ -278,23 +344,55 @@ struct ResizeBilinear<GPUDevice, T> {
     const int total_count = batch * out_height * out_width * channels;
     if (total_count == 0) return;
 
-    GpuLaunchConfig config = GetCudaLaunchConfig(total_count, d);
-    if (half_pixel_centers) {
-      TF_CHECK_OK(CudaLaunchKernel(
-          ResizeBilinearKernel<T>, config.block_count, config.thread_per_block,
-          0, d.stream(), config.virtual_thread_count, images.data(),
-          height_scale, width_scale, batch, in_height, in_width, channels,
-          out_height, out_width, output.data()));
-    } else {
+    // ResizeBilinearKernel_faster is 30 ~ 50% faster than ResizeBilinearKernel
+    // but can only be used when channels is multiple of 4 and size of input
+    // elemnt is the same as float
+    if (channels % 4 == 0 && sizeof(float) == sizeof(T) && half_pixel_centers) {
```

Comment:
I think you need to check T == float.

Reply:
Okay. I wrote a small code which I will include in the bilinear kernel.

The code is:
```cpp
template<typename T>
struct is_float{
  static const bool val = false;
};

template<>
struct is_float<float> {
  static const bool val = true;
};

```

that checks if given type is float or not. I am planning on putting this in a header file: `check_template_type.h` and save it under `tensorflow/core/util/`. Will that work?

Reply:
You can use std::is_same.